### PR TITLE
Introduce QR code generation & serve it via the media server 

### PR DESCRIPTION
**Pull request by siddarthkay**
This PR introduces generating a QR code from status-go side and provides a media server url that can be used to consume it on client side.

The QR code can be generated with the following options :

- URL (this is the url that the QR code will be generated for)
- AllowProfileImage (this is a boolean option, if true is passed it will get your multiaccount profile picture and superimpose it on top of the QR code.
- ErrorCorrectionLevel ( available options are 1,2,3,4 . 1 being lowest error correction and 4 being highest)
- Size (this will be a number and it will generate a QR code of size x size dimensions. Example 100x100 pixels)

### CODE REVIEW

**Comment by ghost** [comment]
### Pull Request Checklist
- [x] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://status.im/docs/))?
- [x] Have you tested changes with mobile?
- [ ] Have you tested changes with desktop?

File: server/handlers.go [hunk 1]
```diff
@@ -384,3 +409,68 @@ func handleIPFS(downloader *ipfs.Downloader, logger *zap.Logger) http.HandlerFun
...
+		qrc, err := qrcode.NewWith(string(qrURLBase64Decoded),
+			qrcode.WithEncodingMode(qrcode.EncModeAuto),
+			qrcode.WithErrorCorrectionLevel(correctionLevel),
+		)
+		if err != nil {
+			fmt.Printf("could not generate QRCode: %v", err)
+		}
+		nw := standard.NewWithWriter(buf)
+		if err = qrc.Save(nw); err != nil {
+			fmt.Printf("could not save image: %v", err)
```
Comment (richard-ramos) [inline server/handlers.go]: Replace fmt.Printf by logger.Error

**Review by richard-ramos — APPROVED** [review]
Cool PR!

**Comment by siddarthkay** [comment]
@status-im/mobile-qa : This is a new feature and I don't think it breaks anything yet, 
I will be requiring this for https://github.com/status-im/status-mobile/issues/14090 and for https://github.com/status-im/status-mobile/issues/13439 (PR coming soon)

**Comment by VolodLytvynenko** [comment]
> @status-im/mobile-qa : This is a new feature and I don't think it breaks anything yet, I will be requiring this for [status-im/status-mobile#14090](https://github.com/status-im/status-mobile/issues/14090) and for [status-im/status-mobile#13439](https://github.com/status-im/status-mobile/issues/13439) (PR coming soon)

@siddarthkay thank you! Let me know when PR is ready for tests

**Comment by siddarthkay** [comment]
@VladimrLitvinenko : cool, So this PR is good to be merged yes?

**Comment by siddarthkay** [comment]
as per : https://github.com/status-im/status-mobile/blob/develop/doc/merging-pr-process.md 
I'm merging this PR now, will tag QA when status-mobile PR is ready :)

**Comment by VolodLytvynenko** [comment]
oh @siddarthkay, I didn't understand what you meant by considering merging right now without testing. It can't break other functionality, but if you don't mind I would like to ask about this my some one else from QA team before merging it

**Comment by siddarthkay** [comment]
Hi @VladimrLitvinenko : sure,
its okay though, I myself was a little bit unsure of the process flow we follow hence I had asked.
So I first checked if all unit tests pass and they do : https://ci.status.im/blue/organizations/jenkins/status-go%2Fprs%2Ftests/detail/PR-3154/4/
and then I checked the process we are following : https://github.com/status-im/status-mobile/blob/develop/doc/merging-pr-process.md
and on top of that this new functionality is no where referenced in `status-mobile` repo so I think QA can start testing when we actually reference this tag in status-mobile :) ( which will be in upcoming PRs)

**Comment by VolodLytvynenko** [comment]
> https://github.com/status-im/status-mobile/blob/develop/doc/merging-pr-process.md

@siddarthkay Got it. Just finished reading the /doc/merging-pr-process.md doc. Thank you for explaining!

**Pull request merged**